src/Sentry/GlobalSessionManager.cs

```diff
@@ -174,7 +183,141 @@ public GlobalSessionManager(SentryOptions options)
+            {
+                _options.DiagnosticLogger?.LogDebug("Persistence directory is not set, returning.");
+                return;
+            }
+
+            try
+            {
+                Directory.CreateDirectory(_persistenceDirectoryPath);
+
+                _options.DiagnosticLogger?.LogDebug(
+                    "Created persistence directory for session file '{0}'.",
+                    _persistenceDirectoryPath
+                );
+
+                var filePath = Path.Combine(_persistenceDirectoryPath, PersistedSessionFileName);
+
+                var persistedSessionUpdate = new PersistedSessionUpdate(update, pauseTimestamp);
+                persistedSessionUpdate.WriteToFile(filePath);
+
+                _options.DiagnosticLogger?.LogInfo(
+                    "Persisted session to a file '{0}'.",
+                    filePath
+                );
+            }
+            catch (Exception ex)
+            {
+                _options.DiagnosticLogger?.LogError(
+                    "Failed to persist session on the file system.",
+                    ex
+                );
+            }
+        }
+
+        private void DeletePersistedSession()
+        {
+            _options.DiagnosticLogger?.LogDebug("Deleting persisted session file.");
+
+            if (string.IsNullOrWhiteSpace(_persistenceDirectoryPath))
+            {
+                _options.DiagnosticLogger?.LogDebug("Persistence directory is not set, returning.");
+                return;
+            }
+
+            try
+            {
+                var filePath = Path.Combine(_persistenceDirectoryPath, PersistedSessionFileName);
+
+                // Try to log the contents of the session file before we delete it
+                if (_options.DiagnosticLogger?.IsEnabled(SentryLevel.Debug) ?? false)
+                {
+                    try
+                    {
+                        _options.DiagnosticLogger?.LogDebug(
+                            "Deleting persisted session file with contents: {0}",
+                            File.ReadAllText(filePath)
+                        );
+                    }
+                    catch (Exception ex)
+                    {
+                        _options.DiagnosticLogger?.LogError(
+                            "Failed to read the contents of persisted session file '{0}'.",
+                            ex,
+                            filePath
+                        );
+                    }
+                }
+
+                File.Delete(filePath);
+
+                _options.DiagnosticLogger?.LogInfo(
+                    "Deleted persisted session file '{0}'.",
+                    filePath
+                );
+            }
+            catch (Exception ex)
+            {
+                _options.DiagnosticLogger?.LogError(
+                    "Failed to delete persisted session from the file system.",
+                    ex
```

Comment:
Given the suggestion above, what about logging the path so we can peek into the filesystem to figure out why:

```suggestion
                _options.DiagnosticLogger?.LogError(
                    "Failed to delete persisted session from the file system: '{0}'",
                    ex,
                    filePath
```